### ISSUE TITLE
some problem about reproducing your results

### ISSUE THREAD

**Issue by Feiyu-Zhang** [xinntao/BasicSR#123]
I use the pre-trained model **RRDB_ESRGAN_x4.pth**. 
When my #epoch reach ~5000, I view generated validation data(Set14). The result is as follow.
![baboon_bicLRx4_265000](https://user-images.githubusercontent.com/34130100/57502572-e3d34e00-731e-11e9-834e-c3bf2166c74e.png)
baboon
![barbara_bicLRx4_265000](https://user-images.githubusercontent.com/34130100/57502600-fd749580-731e-11e9-9776-19a8df5525f7.png)
barbara

Is the result correct at this time? If it is wrong. Is it related to my use of two graphics cards?

**Comment by xinntao** [comment]
There must be some wrong. You need to check your inputs and codes.

**Comment by Feiyu-Zhang** [comment]
> There must be something wrong. You need to check your inputs and codes.
> Two GPUs will not results in the problem, I think.

Thanks for your patient reply. I will check the input and code.

**Comment by Feiyu-Zhang** [comment]
> There must be something wrong. You need to check your inputs and codes.
> Two GPUs will not results in the problem, I think.

I seem to have found the problem. My validate data is 16-bit/color RGB. It seems that data should be 8-bit/color RGB. I think there is a problem during generating LR validate data. I will check _scripts/generate_mod_LR_bic.m_

**Comment by Feiyu-Zhang** [comment]
> There must be something wrong. You need to check your inputs and codes.
> Two GPUs will not results in the problem, I think.

In _scripts/generate_mod_LR_bic.m line:50 img = im2double(img);_ the code in this line will make LR data be 16/bits. I think that is why my problem as described occurs.  I will train again. Hopefully all is well.

**Comment by Feiyu-Zhang** [comment]
> There must be something wrong. You need to check your inputs and codes.
> Two GPUs will not results in the problem, I think.

Hi, xintao. Could you tell me which version you use Matlab/octave image library?

**Comment by xinntao** [comment]
Malab: R2017b (9.3.0.713579) 64-bit (glnxa64) / R2016b (9.1.0.441655) 64-bit (glnxa64) (they produce the same bicubic results)

Opencv: 
```
>>> import cv2
>>> cv2.__version__
'4.0.0'
```

**Comment by Feiyu-Zhang** [comment]
Thanks. Could you tell me which version OpenCV is in Matlab? Because I want
to use *scripts/generate_mod_LR_bic.m *to generate low-resolution images. But
I maybe generate images of other color space. So I want to know your OpenCV
version in Matlab.

**Comment by xinntao** [comment]
I don't think the bicubic function in Matlab will call the opencv library. 
If so, please let me know.

**Comment by Feiyu-Zhang** [comment]
> I don't think the bicubic function in Matlab will call the opencv library.
> If so, please let me know.

I don't use OpenCV in Matlab. I load image-2.10.0 library in Matlab so that I can use imread,im2double,imresize...
But I think my version isn't the same as yours. Could you tell me which **image version** in Matlab?

**Comment by xinntao** [comment]
How can I see the *image version* in my Matlab?

**Comment by Feiyu-Zhang** [comment]
> How can I see the _image version_ in my Matlab?

You are really kind to me. 
I think there are connections with my Octave(replacement of Matlab). It may generate images which are different color space or different bit number.
So I use Matlab to generate LR images eventually. As LR of validation dataset, the model seems to have generated the correct results. As follow,
![baboon_bicLRx4_65000](https://user-images.githubusercontent.com/34130100/57744806-51a3bf00-76fd-11e9-862a-eedde8f95e18.png)
65000 iterations(1300 epochs) PSNR：17.47
Do you think I have started to train correctly?

**Comment by xinntao** [comment]
The learning rate scheme is done in the config file. Wait to see the results (like in 200k iter).

**Comment by Feiyu-Zhang** [comment]
> The learning rate scheme is done in the config file. Wait to see the results (like in 200k iter).

![baboon_265000](https://user-images.githubusercontent.com/34130100/57897871-4d0c1180-7889-11e9-95c5-2cdead88d8b8.png)
Above is baboon which is trained 265000 iterations. Do you think where is the problem?

**Comment by xinntao** [comment]
It seems strange.... 1) may check your data 2) do you use the default config file?

**Comment by shiyangjing** [comment]
> > There must be something wrong. You need to check your inputs and codes.
> > Two GPUs will not results in the problem, I think.
> 
> In _scripts/generate_mod_LR_bic.m line:50 img = im2double(img);_ the code in this line will make LR data be 16/bits. I think that is why my problem as described occurs. I will train again. Hopefully all is well.

How did you solve the problem, by modifying the function im = im2double(img)?

**Comment by Feiyu-Zhang** [comment]
> > > There must be something wrong. You need to check your inputs and codes.
> > > Two GPUs will not results in the problem, I think.
> > 
> > 
> > In _scripts/generate_mod_LR_bic.m line:50 img = im2double(img);_ the code in this line will make LR data be 16/bits. I think that is why my problem as described occurs. I will train again. Hopefully all is well.
> 
> How did you solve the problem, by modifying the function im = im2double(img)?

I run code in Matlab finally. It succeeded.
